### ISSUE TITLE
[validation] missing alt text not properly detected if paired with an attribute

### ISSUE THREAD

**Issue by zkamvar** [carpentries/pegboard#106]
In https://github.com/fishtree-attempt/instructor-training/pull/4, @dpshelio and @karenword attempted to remove alt text to trigger a failing check, but nothing failed. 

I do not know the exact mechanism right now, but when I replicate this, I see that the alt text is coded as the character "NA":

``` r
tmp <- withr::local_tempfile()
writeLines(c("test", "test two", "![test image](https://example.com/img.jpg){width='50%'}"), tmp)
pegboard::Episode$new(tmp)$validate_links() |> as.list()
#> $scheme
#> [1] "https"
#> 
#> $server
#> [1] "example.com"
#> 
#> $port
#> [1] NA
#> 
#> $user
#> [1] ""
#> 
#> $path
#> [1] "/img.jpg"
#> 
#> $query
#> [1] ""
#> 
#> $fragment
#> [1] ""
#> 
#> $orig
#> [1] "https://example.com/img.jpg"
#> 
#> $text
#> [1] "test image"
#> 
#> $alt
#> [1] "NA"
#> 
#> $title
#> [1] ""
#> 
#> $type
#> [1] "image"
#> 
#> $rel
#> [1] NA
#> 
#> $anchor
#> [1] FALSE
#> 
#> $sourcepos
#> [1] 3
#> 
#> $filepath
#> file19dda19322d4e
#> 
#> $node
#> [[1]]
#> {xml_node}
#> <image sourcepos="3:1-3:42" destination="https://example.com/img.jpg" title="" alt="NA">
#> [1] <text sourcepos="3:3-3:12" xml:space="preserve">test image</text>
#> 
#> 
#> $enforce_https
#> [1] TRUE
#> 
#> $internal_anchor
#> [1] TRUE
#> 
#> $internal_file
#> [1] TRUE
#> 
#> $internal_well_formed
#> [1] TRUE
#> 
#> $all_reachable
#> [1] TRUE
#> 
#> $img_alt_text
#> [1] TRUE
#> 
#> $descriptive
#> [1] TRUE
#> 
#> $link_length
#> [1] TRUE
```

<sup>Created on 2023-01-23 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by zkamvar** [comment]
The alt text is validated in 

https://github.com/carpentries/pegboard/blob/51d5b5e887b7d178be2915d8e60171f352dcf632/R/validate_links.R#L114-L120

Funny enough, the alt text is provisioned here and the fatal mistake is _assuming that all images with attributes will contain an alt text element_ 🤦🏼 :

https://github.com/carpentries/pegboard/blob/51d5b5e887b7d178be2915d8e60171f352dcf632/R/get_images.R#L95

**Comment by zkamvar** [comment]
In #108 this works

``` r
tmp <- withr::local_tempfile()
writeLines(c("test", "test two", "![test image](https://example.com/img.jpg){width='50%'}"), tmp)
pegboard::Episode$new(tmp)$validate_links() |> as.list()
#> ! There were errors in 1/1 links
#> ◌ Images need alt-text
#> <https://webaim.org/techniques/hypertext/link_text#alt_link>
#> 
#> file75ba9e1abf5:3 [image missing alt-text]
#> $scheme
#> [1] "https"
#> 
#> $server
#> [1] "example.com"
#> 
#> $port
#> [1] NA
#> 
#> $user
#> [1] ""
#> 
#> $path
#> [1] "/img.jpg"
#> 
#> $query
#> [1] ""
#> 
#> $fragment
#> [1] ""
#> 
#> $orig
#> [1] "https://example.com/img.jpg"
#> 
#> $text
#> [1] "test image"
#> 
#> $alt
#> [1] NA
#> 
#> $title
#> [1] ""
#> 
#> $type
#> [1] "image"
#> 
#> $rel
#> [1] NA
#> 
#> $anchor
#> [1] FALSE
#> 
#> $sourcepos
#> [1] 3
#> 
#> $filepath
#> file75ba9e1abf5
#> 
#> $node
#> [[1]]
#> {xml_node}
#> <image sourcepos="3:1-3:42" destination="https://example.com/img.jpg" title="">
#> [1] <text sourcepos="3:3-3:12" xml:space="preserve">test image</text>
#> 
#> 
#> $enforce_https
#> [1] TRUE
#> 
#> $internal_anchor
#> [1] TRUE
#> 
#> $internal_file
#> [1] TRUE
#> 
#> $internal_well_formed
#> [1] TRUE
#> 
#> $all_reachable
#> [1] TRUE
#> 
#> $img_alt_text
#> [1] FALSE
#> 
#> $descriptive
#> [1] TRUE
#> 
#> $link_length
#> [1] TRUE
```

<sup>Created on 2023-01-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by zkamvar** [comment]
In #108 this works

``` r
tmp <- withr::local_tempfile()
writeLines(c("test", "test two", "![test image](https://example.com/img.jpg){width='50%'}"), tmp)
pegboard::Episode$new(tmp)$validate_links() |> as.list()
#> ! There were errors in 1/1 links
#> ◌ Images need alt-text
#> <https://webaim.org/techniques/hypertext/link_text#alt_link>
#> 
#> file75ba9e1abf5:3 [image missing alt-text]
#> $scheme
#> [1] "https"
#> 
#> $server
#> [1] "example.com"
#> 
#> $port
#> [1] NA
#> 
#> $user
#> [1] ""
#> 
#> $path
#> [1] "/img.jpg"
#> 
#> $query
#> [1] ""
#> 
#> $fragment
#> [1] ""
#> 
#> $orig
#> [1] "https://example.com/img.jpg"
#> 
#> $text
#> [1] "test image"
#> 
#> $alt
#> [1] NA
#> 
#> $title
#> [1] ""
#> 
#> $type
#> [1] "image"
#> 
#> $rel
#> [1] NA
#> 
#> $anchor
#> [1] FALSE
#> 
#> $sourcepos
#> [1] 3
#> 
#> $filepath
#> file75ba9e1abf5
#> 
#> $node
#> [[1]]
#> {xml_node}
#> <image sourcepos="3:1-3:42" destination="https://example.com/img.jpg" title="">
#> [1] <text sourcepos="3:3-3:12" xml:space="preserve">test image</text>
#> 
#> 
#> $enforce_https
#> [1] TRUE
#> 
#> $internal_anchor
#> [1] TRUE
#> 
#> $internal_file
#> [1] TRUE
#> 
#> $internal_well_formed
#> [1] TRUE
#> 
#> $all_reachable
#> [1] TRUE
#> 
#> $img_alt_text
#> [1] FALSE
#> 
#> $descriptive
#> [1] TRUE
#> 
#> $link_length
#> [1] TRUE
```

<sup>Created on 2023-01-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
